src/components/CalendarGrid.vue

```diff
@@ -154,6 +168,16 @@ export default {
 			}
 		},
 		eventSources() {
+			if (this.isWidget) {
+				const publicUrl = `/remote.php/dav/public-calendars/${this.token}/`
+				const url = this.url
```

Comment:
```suggestion
				const publicUrl = OC.webroot + `/remote.php/dav/public-calendars/${this.token}/`
				const url = OC.webroot + this.url
```

Reply:
I'm not sure where the url is generated as I'm not that familiar with the widget backend stuff.

I guess the url should be fixed in the backend where it is generated.

Reply:
url comes from the pasted text, in Talk , text ...

Reply:
I just read Christoph's comment and there is a cleaner variant available: https://nextcloud-libraries.github.io/nextcloud-router/functions/getRootUrl.html

So it should be `getRootUrl()` instead of `OC.webroot`.

Reply:
> url comes from the pasted text, in Talk , text ...

Yeah, I'm confused because I pasted the full url with `https://` and the webroot into Talk. Maybe this bug affects other widgets as well.

Reply:
Oh, yes what we get in the frontend is extracted using regex in `lib/Reference/ReferenceProvider.php` because calendar's in the store looks like this 
<img width="725" alt="image" src="https://github.com/nextcloud/calendar/assets/40746210/9eab9c41-9370-4282-b4b0-f1216422b210">

Reply:
can be added in the backend using `getWebroot` from `IURLGenerator` that way we don't event have to check for `isPublic`